src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -545,8 +545,14 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                case 'tagged_iterator':
                 case 'tagged_locator':
                     $type = $arg->getAttribute('type');
+
+                    if ('tagged' === $type) {
+                        @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated and will be removed in Symfony 5. Please use "tagged_iterator"', $type, $name, $file), E_USER_DEPRECATED);
```

Comment:
no need for the "if": this should be moved in the "case" above

Reply:
I might be wrong, but wouldn't this mean, that the tag `!tagged` will not work anymore? He will see the deprecation notice, but we force him to use the new tag `!tagged_iterator`.